pkg/pillar/worker/worker.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// worker is used to kick off some work to a goroutine and get a notification
+// when the work is complete
+
+package worker
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Worker captures the worker channels
+type Worker struct {
+	// Private
+	requestChan  chan<- Work
+	resultChan   <-chan privateResult
+	requestCount uint
```

Comment:
it would be great to provide brief comments for requestCount/resultCount semantics

Reply:
Will do

Reply:
Done